script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -110,50 +134,231 @@ class PublishPluginCommand extends PluginCommand {
       remoteUrl = await _verifyRemote(remote);
     }
     _print('Local repo is ready!');
+    if (argResults[_dryRunFlag] as bool) {
+      _print('===============  DRY RUN ===============');
+    }
+
+    bool successful;
+    if (publishAllChanged) {
+      successful = await _publishAllChangedPackages(
+        remote: remote,
+        remoteUrl: remoteUrl,
+        shouldPushTag: shouldPushTag,
+        baseGitDir: baseGitDir,
+      );
+    } else {
+      successful = await _publishAndTagPackage(
+        packageDir: _getPackageDir(package),
+        remote: remote,
+        remoteUrl: remoteUrl,
+        shouldPushTag: shouldPushTag,
+      );
+    }
+    await _finish(successful);
+  }
+
+  Future<bool> _publishAllChangedPackages({
+    String remote,
+    String remoteUrl,
+    bool shouldPushTag,
+    GitDir baseGitDir,
+  }) async {
+    final GitVersionFinder gitVersionFinder = await retrieveVersionFinder();
+    final List<String> changedPubspecs =
+        await gitVersionFinder.getChangedPubSpecs();
+    if (changedPubspecs.isEmpty) {
+      _print('No version updates in this commit.');
+      return true;
+    }
+    _print('Getting existing tags...');
+    final io.ProcessResult existingTagsResult =
+        await baseGitDir.runCommand(<String>['tag', '--sort=-committerdate']);
+    final List<String> existingTags = (existingTagsResult.stdout as String)
+        .split('\n')
+          ..removeWhere((String element) => element.isEmpty);
+
+    final List<String> packagesReleased = <String>[];
+    final List<String> packagesFailed = <String>[];
+
+    for (final String pubspecPath in changedPubspecs) {
+      final File pubspecFile =
+          fileSystem.directory(baseGitDir.path).childFile(pubspecPath);
+      final _CheckNeedsReleaseResult result = await _checkNeedsRelease(
+        pubspecFile: pubspecFile,
+        gitVersionFinder: gitVersionFinder,
+        existingTags: existingTags,
+      );
+      switch (result) {
+        case _CheckNeedsReleaseResult.release:
+          break;
+        case _CheckNeedsReleaseResult.noRelease:
+          continue;
+        case _CheckNeedsReleaseResult.failure:
+          packagesFailed.add(pubspecFile.parent.basename);
+          continue;
+      }
+      _print('\n');
+      if (await _publishAndTagPackage(
+        packageDir: pubspecFile.parent,
+        remote: remote,
+        remoteUrl: remoteUrl,
+        shouldPushTag: shouldPushTag,
+      )) {
+        packagesReleased.add(pubspecFile.parent.basename);
+      } else {
+        packagesFailed.add(pubspecFile.parent.basename);
+      }
+      _print('\n');
+    }
+    if (packagesReleased.isNotEmpty) {
+      _print('Packages released: ${packagesReleased.join(', ')}');
+    }
+    if (packagesFailed.isNotEmpty) {
+      _print(
+          'Packages failed to release: ${packagesFailed.join(', ')}, see above for details.');
```

Comment:
Nit: "Failed to release the following packages:"

Reply:
done